nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/registry/flow/mapping/NiFiRegistryFlowMapper.java

```diff
@@ -599,23 +635,35 @@ private BatchSize mapBatchSettings(final RemoteGroupPort remotePort) {
         return batchSize;
     }
 
-    public VersionedParameterContext mapParameterContext(final ParameterContext context) {
-        if (context == null) {
-            return null;
-        }
-
-        final Set<VersionedParameter> parameters = context.getParameters().values().stream()
-            .map(this::mapParameter)
-            .collect(Collectors.toSet());
+    public Collection<VersionedParameterContext> mapParameterContexts(final ProcessGroup processGroup, final boolean mapDescendantVersionedFlows) {
```

Comment:
Am thinking out loud a bit here. I don't yet fully understand the entire context around where this will be used. But it seems to me anywhere that we would want a Collection of VersionedParameterContext it would probably make sense to restrict it to a `Set<VersionedParameterContext>`. Because we can certainly have the case where descendant Process Groups will reference the same ParameterContext as a parent/ancestor Process Group. I don't think we would actually want the context in the Collection multiple times, would we?

Reply:
Agreed, moving everything to a Map to begin with, as we discussed